go/vt/sqlparser/ast.go

```diff
@@ -1720,8 +1731,9 @@ func (a ReferenceAction) Format(buf *TrackedBuffer) {
 	}
 }
 
-// ForeignKeyDefinition describes a foreign key in a CREATE TABLE statement
+// ForeignKeyDefinition describes a foreign key
 type ForeignKeyDefinition struct {
+	IsDrop            bool
```

Comment:
Think I would prefer this to be an action string, as in other DDL / column actions

Reply:
Thinking about this a little further, I don't think this belongs in the definition struct at all. This should be a "constraintAction" field in the parent DDL action struct. Column modifications / drops / adds have a similar structure you should copy.